packages/salesforcedx-vscode-apex/src/requirements.ts

```diff
@@ -113,7 +113,7 @@ export const checkJavaVersion = async (javaHome: string): Promise<boolean> => {
           )
         );
       }
-      if (!/java\.version\s*=\s*(?:11|17)/g.test(stderr)) {
+      if (!/java\.version\s*=\s*(?:11|17|21)/g.test(stderr)) {
```

Comment:
@diyer I believe the guard should be on the the lowest supported version. Given LSP is targeted at a minimum java version at compile time any java version that >= min should work fine. The other compile time constraint is source code version. For 250, this would be java 11 and for 252 java 17. So using new language feature that appears in a new release of java cannot be used until language version includes the version where the feature appears.